wasm/jit/jit_amd64.go

```diff
@@ -1182,6 +1169,122 @@ func (b *amd64Builder) handleLoad(o *wazeroir.OperationLoad) error {
 	return nil
 }
 
+func (b *amd64Builder) handleLoad8(o *wazeroir.OperationLoad8) error {
+	base := b.locationStack.pop()
+	if err := b.ensureLocatedOnGeneralPurposeRegister(base); err != nil {
+		return err
+	}
+
+	// At this point, base's value is on the integer general purpose reg.
+	// We reuse the register below, so we alias it here for readability.
+	reg := base.register
+
+	// We have to calculate the offset on the memory region.
+	addOffsetToBase := b.newProg()
+	addOffsetToBase.As = x86.AADDL // 32-bit!
+	addOffsetToBase.To.Type = obj.TYPE_REG
+	addOffsetToBase.To.Reg = reg
+	addOffsetToBase.From.Type = obj.TYPE_CONST
+	addOffsetToBase.From.Offset = int64(o.Arg.Offest)
+	b.addInstruction(addOffsetToBase)
+
+	// Then move a byte at the offset to the register.
+	// Note that Load8 is only for integer types.
+	moveFromMemory := b.newProg()
+	moveFromMemory.As = x86.AMOVB // 1 byte!
+	moveFromMemory.To.Type = obj.TYPE_REG
+	moveFromMemory.To.Reg = reg
+	moveFromMemory.From.Type = obj.TYPE_MEM
+	moveFromMemory.From.Reg = reservedRegisterForMemory
+	moveFromMemory.From.Index = reg
+	moveFromMemory.From.Scale = 1
+	b.addInstruction(moveFromMemory)
+	top := b.locationStack.pushValueOnRegister(reg)
+
+	switch o.Type {
```

Comment:
this deserves a comment (why the top type is unsigned even if the value type is signed)